pkg/crypto/clef/clef.go

```diff
@@ -66,17 +67,39 @@ func DefaultIpcPath() (string, error) {
 	return filepath.Join(home, ".clef", socket), nil
 }
 
+func selectAccount(clef ExternalSignerInterface, ethAddress *common.Address) (accounts.Account, error) {
+	// get the list of available ethereum accounts
+	clefAccounts := clef.Accounts()
+	if len(clefAccounts) == 0 {
+		return accounts.Account{}, ErrNoAccounts
+	}
+
+	if ethAddress == nil {
+		// pick the first account as the one we use
+		return clefAccounts[0], nil
+	}
+
+	for _, availableAccount := range clefAccounts {
+		if availableAccount.Address == *ethAddress {
+			return availableAccount, nil
+		}
+	}
+	return accounts.Account{}, ErrAccountNotAvailable
+}
+
 // NewSigner creates a new connection to the signer at endpoint.
 // As clef does not expose public keys it signs a test message to recover the public key.
-func NewSigner(clef ExternalSignerInterface, client Client, recoverFunc crypto.RecoverFunc) (signer crypto.Signer, err error) {
+func NewSigner(clef ExternalSignerInterface, client Client, recoverFunc crypto.RecoverFunc, ethAddress *common.Address) (signer crypto.Signer, err error) {
 	// get the list of available ethereum accounts
 	clefAccounts := clef.Accounts()
 	if len(clefAccounts) == 0 {
 		return nil, ErrNoAccounts
 	}
```

Comment:
Technically, this is not needed now, as you do this check in the function `selectAccount`.

Reply:
removeed